File: music-application/src/main/java/com/music/sale/application/category/port/inport/CategoryQueryUseCase.java
// Copyright (C) 2024 Your Name or Company
package com.music.sale.application.category.port.inport;

import com.music.sale.domain.category.Category;
import com.music.sale.domain.category.enums.CategoryType;
import java.util.List;

public interface CategoryQueryUseCase {
  Category getCategoryById(Long id);

  List<Category> getAllCategories();

  List<Category> getCategoriesByType(CategoryType type);

  List<Category> getRootCategories();

  List<Category> getCategoriesByParentId(Long parentId);
}

File: music-application/src/main/kotlin/com/music/sale/application/category/port/inport/CategoryQueryUseCase.kt
// Copyright (C) 2024 Your Name or Company
package com.music.sale.application.category.port.inport

import com.music.sale.domain.category.Category
import com.music.sale.domain.category.CategoryType

interface CategoryQueryUseCase {
    fun getCategoryById(id: Long): Category

    fun getAllCategories(): List<Category>

    fun getCategoriesByType(type: CategoryType): List<Category>

    fun getRootCategories(): List<Category>

    fun getCategoriesByParentId(parentId: Long): List<Category>
}
